compiler/rustc_typeck/src/check/demand.rs

```diff
@@ -857,14 +839,17 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                                 Applicability::MachineApplicable,
                                 false,
                             ));
-                        } else if self.infcx.type_is_copy_modulo_regions(
-                            self.param_env,
-                            expected,
-                            sp,
-                        ) {
-                            // For this suggestion to make sense, the type would need to be `Copy`.
+                        }
+
+                        // For this suggestion to make sense, the type would need to be `Copy`,
+                        // or we have to be moving out of a `Box<T>`
+                        if self.infcx.type_is_copy_modulo_regions(self.param_env, expected, sp)
+                            || checked_ty.is_box()
+                        {
                             if let Ok(code) = sm.span_to_snippet(expr.span) {
-                                let message = if checked_ty.is_region_ptr() {
+                                let message = if checked_ty.is_box() {
+                                    "consider unboxing the value"
```

Comment:
I kind of like the new "consider unboxing" wording, but it seems inconsistent with the other branches, especially the "consider dereferencing the type". What's your thinking about that?

Reply:
I have no opinion on the matter. I could change this to say `consider unboxing the type` so it's a bit more consistent?

Reply:
Hmm, in some ways `consider unboxing the type` seems confusing, because we're not saying to remove `Box<...>` from a type. Let's just leave it as `consider unboxing the value`.

Reply:
yeah. tempted to change the other messages to say "value" as well, but that's out of the scope of this PR.